scripts/checker.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+set -e
+set -u
+set -o pipefail
+
+
+check_exit_code() {
+    exit_code=$?
+    # full path to output file
+    #local checker_file="/var/lib/container_probes/checker_ready.txt"
+
+    # analyze exit code with trapped function
+    if [ ${exit_code} -ne 0 ]; then
+        # Clear the readiness file
+        echo "ERROR: Liveness probe failed with code ${exit_code}" # | tee "${checker_file}"
+        exit_code=1
+    else
+        # TODO: implement robust readiness check
+        #./checker_ready.sh
+        echo "READY" # | tee ${checker_file}
+    fi
+    exit ${exit_code}
+}
+
+# Capture Exit Code with trapped pseudo signal EXIT
+# When the script is completed or exits for any reason, the
+# commands in the check_exit_code function will be executed.
+trap check_exit_code EXIT
+
+# Do the lifeness check
+searchd --status 1> /dev/null
```

Comment:
What if `searchd --status` is working but returns an error ?

I would also add at the end of the script a `exit 0` just to be clear that script completion means everything is OK.

Reply:
> What if searchd --status is working but returns an error ?
dont see any use case how that could happen. if searchd is running and accepting connections searchd returns 0. in any other case it returns an error code. the container can be considered running if searchd is running (exit 0)

the exit code is generated in the trap function `check_exit_code` which is called on EXIT signal.  so the exit code is always 1 or 0.